corehq/apps/users/models.py

```diff
@@ -3380,14 +3381,23 @@ class ConnectIDUserLink(models.Model):
     class Meta:
         unique_together = ('domain', 'commcare_user')
 
-    @property
+    @cached_property
     def messaging_key(self):
         key = generate_aes_key().decode("utf-8")
         messaging_key, _ = ConnectIDMessagingKey.objects.get_or_create(
             connectid_user_link=self, domain=self.domain, active=True, defaults={"key": key}
         )
         return messaging_key
 
+    @cached_property
+    def messaging_channel(self):
+        # inline to prevent circular import
+        from corehq.messaging.smsbackends.connectid.backend import ConnectBackend
+
+        if self.channel_id is None:
+            ConnectBackend().create_channel(self)
```

Comment:
Not new in this PR, but I think it'd be more typical to make `create_channel` a static method.